pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,268 @@
+		return nil, fmt.Errorf("failed to open connection: %w", err)
+	}
+
+	p := &Provider{
+		dbURL: dbPath,
+		db:    db,
+		dbs:   map[string]*sqlDBStore{}}
+
+	return p, nil
+}
+
+// OpenStore opens and returns new db for given name space.
+func (p *Provider) OpenStore(name string) (storage.Store, error) {
+	p.Lock()
+	defer p.Unlock()
+
+	// check name is not empty
+	if name == "" {
+		return nil, errors.New("store name is required")
+	}
+	// creating the database
+	_, err := p.db.Exec(createDBQuery + name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create db %s: %w", name, err)
+	}
+
+	// Opening new db connection
+	newDBConn, err := sql.Open("mysql", p.dbURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new connection %s: %w", p.dbURL, err)
+	}
+
+	// Use is used to select the created database without this DDL operations are not permitted
+	_, err = newDBConn.Exec(useDBQuery + name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to use db %s: %w", name, err)
+	}
+
+	tableName := tablePrefix + name
+	// key has max varchar size it can accommodate as per mysql 8.0 spec
+	createTableStmt := "CREATE Table IF NOT EXISTS " + tableName +
+		"(`key` varchar(255) NOT NULL ,`value` BLOB, PRIMARY KEY (`key`));"
+
+	// creating key-value table inside the database
+	_, err = newDBConn.Exec(createTableStmt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create table %s: %w", tableName, err)
+	}
+
+	store := &sqlDBStore{
+		db:        newDBConn,
+		tableName: tableName}
+
+	p.dbs[name] = store
+
+	return store, nil
+}
+
+// Close closes the provider.
+func (p *Provider) Close() error {
+	p.Lock()
+	defer p.Unlock()
+
+	for _, store := range p.dbs {
+		err := store.db.Close()
+		if err != nil {
+			return fmt.Errorf(failToCloseProviderErrMsg+": %w", err)
+		}
+	}
+
+	if err := p.db.Close(); err != nil {
+		return err
+	}
+
+	p.dbs = make(map[string]*sqlDBStore)
+
+	return nil
+}
+
+// CloseStore closes a previously opened store
+func (p *Provider) CloseStore(name string) error {
+	p.Lock()
+	defer p.Unlock()
+
+	store, exists := p.dbs[name]
+	if !exists {
+		return nil
+	}
+
+	delete(p.dbs, name)
+
+	return store.db.Close()
+}
+
+// Put stores the key and the record
+func (s *sqlDBStore) Put(k string, v []byte) error {
+	if k == "" {
+		return errors.New("key is mandatory")
+	}
+
+	//nolint: gosec
+	// create upsert query to insert the record, checking whether the key is already mapped to a value in the store.
+	createStmt := "INSERT INTO " + s.tableName + " VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?"
+	// executing the prepared insert statement
+	_, err := s.db.Exec(createStmt, k, v, v)
+	if err != nil {
+		return fmt.Errorf("failed to insert key and value record into %s %w ", s.tableName, err)
+	}
+
+	return nil
+}
+
+// Get fetches the record based on key
+func (s *sqlDBStore) Get(k string) ([]byte, error) {
+	if k == "" {
+		return nil, errors.New("key is mandatory")
+	}
+
+	var value []byte
+	//nolint: gosec
```

Comment:
Similar to one of my comments above - `gosec` warning about SQL injection attack?

Reply:
See if https://www.calhoun.io/what-is-sql-injection-and-how-do-i-avoid-it-in-go/ is helpful

Reply:
Not really 
You see we have users here parameterized, you will barely find the example where table name which is aka store name for us is variable 
db.Exec("INSERT INTO users(name, email) VALUES($1, $2)",
  "Jon Calhoun", "jon@calhoun.io")

Reply:
Perhaps you'll have to manually write the escaped SQL if the library doesn't have a way to allow parameterized table names?

Reply:
@DRK3 https://github.com/go-sql-driver/mysql/issues/848

I guess @talwinder50 can confirm if the same issue exists for SELECT and INSERT queries.

Reply:
Yes @llorllale  issue exists for select and insert queries too

Reply:
Yeah, I'm guessing the linter can't tell if you've properly sanitized the input, so the gosec override will probably have to remain even after you've fixed it